src/transformer.js

```diff
@@ -1,18 +1,30 @@
 import { execSync } from 'child_process'
-import { basename } from 'path'
+import { basename, extname } from 'path'
 import { pathToFileURL } from 'url'
 import * as svelte from 'svelte/compiler'
 
 import { getSvelteConfig } from './svelteconfig.js'
 
-const dynamicImport = async (filename) => import(pathToFileURL(filename).toString())
+const dynamicImport = async (filename) => import(pathToFileURL(filename))
+
+const currentFileExtension = (global.__dirname !== undefined ? extname(__filename) : extname(pathToFileURL(import.meta.url).toString())).replace('.', '')
+
+const inCommonJs = typeof module !== 'undefined'
 
 /**
  * Jest will only call this method when running in ESM mode.
  */
 const processAsync = async (source, filename, jestOptions) => {
   const options = jestOptions && jestOptions.transformerConfig ? jestOptions.transformerConfig : {}
-  const { preprocess, rootMode } = options
+  const { preprocess, rootMode, debug } = options
+
+  if (inCommonJs) {
+    throw new Error('Running svelte-jester-transformer async in unsupported CJS mode')
+  }
+
+  if (debug) {
+    console.debug(`Running svelte-jester-transformer async in mode ${currentFileExtension}.`)
+  }
 
   if (!preprocess) {
     return compiler('esm', options, filename, source)
```

Comment:
we need to omit the `format` option with Svelte 4 or else users will get errors printed to the console. I think we can do something like `import { VERSION } from 'svelte';` to check what version users are on. `@sveltejs/vite-plugin-svelte` does this